docs: Remove the +New option for creating new models

eagereyes opened the pull request:
Adapts the instructions for creating new models to no longer mention the use of the +New button.

This is the only place in the docs I could find where the New button is explicitly mentioned. I'll make a separate PR for a page on Learn in the other repo.

eagereyes committed:
Remove the +New option for creating new models

alexyarosh reviewed (changes requested):
@eagereyes please mention that one could create a model from scratch by going to models in the nav sidebar and clicking the + button
![Screenshot_20250922_190216~2](https://github.com/user-attachments/assets/5d1ae06d-b108-480f-93c1-0db4238c9821)

Also might be worth mentioning the route of "command palette  > new model" (see metrics docs for an example)

eagereyes committed:
Bring back the from-scratch approach

Based on Alex's requested change, but also realized that you can use the command palette

eagereyes committed:
Move note about models getting pinned back into the new model section

eagereyes committed:
Format UI terms using bold instead of italics

alexyarosh reviewed (approved):
Thank you!
